### DISCUSSION TITLE
Beginner - FCP Keyboard Shortcut for Inspector Rotation of clip?

### DISCUSSION THREAD

**Discussion by spdorsey** [CommandPost/CommandPost#3133] [Technical Support] [answered]
Hello

  I'm new to CommandPost. It looks pretty cool, but I don't know it very well yet. I have been online watching YouTube videos and scouring the support documentation and I have not found a solution to my issue. 

  I often need to rotate footage by 90º or 180º. I would LOVE to have a hotkey (command-F9, for instance) to accomplish this without needing to go to the Inspector inside FCP for each clip or as a batch. 

  Is there a way to set this up using CommandPost? 

Thanks very much.

**Comment by randomeizer** [comment]
@latenitefilms may know of an existing shortcut that does this on a single clip, but for a batch I suspect it would require a custom plugin. The good news is, it's not a complicated plugin, but will require a bit of Lua code.

I don't have my computer in front of me at the moment, but one of us can probably put something together in the not-too-distant future.

One thing to note is that as of 10.6.4, there was a selection bug in the timeline where selecting Secondary Story clips will wipe the list of selected clips (from a code point of view). Haven't had a chance to test if it has been fixed in the release overnight.

**Comment by latenitefilms** [comment, marked as answer]
Apologies for the delayed reply. The easiest way to do this is with a Lua Snippet.

From the menubar select **Preferences > Scripting**:

<img width="425" alt="image" src="https://user-images.githubusercontent.com/22286696/197936436-5df9ae5d-57e7-45ec-a333-df56c758b279.png">

Click the **New** button then type a name.

Paste in this code (changing the 90 to whatever):

```lua
cp.apple.finalcutpro.inspector.video:transform():rotation():show():value(90)
```

You can now access your Snippet through the Search Console, and assign it to a keyboard shortcut or control surface button.

<img width="1112" alt="image" src="https://user-images.githubusercontent.com/22286696/197936582-6abb0ddb-50d4-42ba-9e2e-8804e8c47b37.png">

<img width="1392" alt="image" src="https://user-images.githubusercontent.com/22286696/197936627-aa7593e5-1532-4e12-a7eb-c98a82cec4f7.png">

**Comment by spdorsey** [comment]
Thanks very much for your reply. This makes it easy for me to make a variety of controls for often-used actions!

**Comment by DaniDitu** [comment]
Condivido i miei codici LUA snippet, se a qualcuno può aiutare: 
cp.apple.finalcutpro:launch() -- apre Final cut
cp.apple.finalcutpro:launch().inspector.video:show() -- apre l'ispettore video delle clip selezionato
cp.apple.finalcutpro.inspector.video:transform():rotation():show():value(90) -- ruota di 90
cp.apple.finalcutpro.inspector.video:compositing():opacity():show():value(50) -- imposta l'opacità del clip al 50%

cp.apple.finalcutpro:launch():timeline():show():selectClip()
cp.apple.finalcutpro:launch():selectMenu({"Window", "Go To", "Timeline"}) -- ti porta nella timeline
hs.eventtap.keyStroke({},"c", 0) -- esempio di keystroke select clip

cp.apple.finalcutpro:launch():selectMenu({"Window", "Go To", "Timeline"}) -- ti porta nella timeline
cp.triggerAction("fcpx_cmds",{id="cpSelectClipAtLaneOne"}) -- seleziona la clip in corrispondenza del playhead

cp.apple.finalcutpro:launch():selectMenu({"Edit", "Deselect All"}) --deselect all clip in timeline
cp.apple.finalcutpro:launch():selectMenu({"Edit", "Select Clip"}) --Select Clip in Timeline

-- Copy attribute and Paste to Next Clip
cp.apple.finalcutpro:launch()
hs.eventtap.keyStroke({"cmd", ""}, "2") --go to timelene
hs.eventtap.keyStroke({},"c")--select clip
hs.eventtap.keyStroke({"cmd"}, "c") --copy
hs.eventtap.keyStroke({"shift", "cmd"}, "a") --deselect (scorciatoia aggiunta a FCPX)
hs.eventtap.keyStroke({},"down", "right") --go to next clip sposta il playhead avanti
hs.eventtap.keyStroke({"shift"}, "right") --sposta di 10 frame
hs.eventtap.keyStroke({"shift"}, "right") --sposta di 10 frame
hs.eventtap.keyStroke({"shift"}, "right") --sposta di 10 frame
hs.eventtap.keyStroke({"shift"}, "right") --sposta di 10 frame
cp.just.wait(0.2) -- Wait 1 second
hs.eventtap.keyStroke({"ctrl", "shift"}, "v", 0.5) --paste
hs.eventtap.keyStroke({}, "return")
--
